### ISSUE TITLE
Some users see information they shouldn't see

### ISSUE THREAD

**Issue by anditee** [microsoft-search/pnp-modern-search#3872]
**Version used**
4.12.2

**Describe the bug**
We are using the webpart to enable search for different articles (sitepages) within our sharepoint site. The solution works fine for almost all of our customers and all our users in the company.

Some of the customers experience a weird error depending issues with the results shown. At first glance we thought about an issue regarding the access rights of the user itself but testing it in SearchQueryTool by Microsoft doesn't show the same results as what we are experiencing within the webpart results.

Query Text used:
{searchTerms}* 
ContentTypeId:{0x0101009D1CB255DA76424F860D91F20E6C411800922E4C6D4DC3A54F909869CDB204094B*} ContentClass:"STS_ListItem_WebPageLibrary"

What user with the error gets shown:
He can see everything for example other users names and e-mail adresses, draft versions, files he has only read access to. (about 70 files and all users)

What user without the error gets shown:
The corresponding documents with the right contentTypeID and he is able to filter (about 50 files)

**To Reproduce**
The only thing we noticed is that its on some users regardless if guest user or tenant user or admin.

**Expected behavior**
The experience should be equal for every user and only include the right content type. Furthermore the query text should apply filters to the search.

**Desktop (please complete the following information):**
 - all modern browsers

**Additional context**
Customer get access via cross tenant sync

**Comment by kasperbolarsen** [comment]
I assume you have tested if the user that are "seeing too much" actually can see the content,  following the link to the content? Getting an Access Denied?

**Comment by anditee** [comment]
you are right for example if he clicks on users which he sees which shouldn't be shown he gets an access denied.
